src/annot/mod.rs

```diff
@@ -0,0 +1,87 @@
+//! Data types for positions and regions on named sequences
+//! (e.g. chromosomes), useful for annotating features in a genome.
+
```

Comment:
Please add a doc comment here that guides people through the entire module. What can be done, what different structures are there and how can they be used. Including examples.

Reply:
I expanded this quite a bit and added example code